File: Source/Entities/SampleActor.vb
Imports Celeste.[Mod].Entities
Imports Microsoft.Xna.Framework

<CustomEntity("VeryBasicMod/SampleActor")>
Public Class SampleActor
    Inherits Actor
    Public Sub New(data As EntityData, offset As Vector2)
        MyBase.New(data.Position + offset)
        ' TODO: read properties from data
    End Sub

End Class

File: Source/Entities/SampleActor.fs
namespace Celeste.Mod.VeryFunnyMod.Entities

open Celeste.Mod
open Celeste
open System
open Microsoft.Xna.Framework
open Celeste.Mod.Entities

[<CustomEntity("VeryFunnyMod/SampleActor")>]
type SampleActor(data: EntityData, offset: Vector2) as this =
    inherit Actor(data.Position + offset)
